src/main/java/muffin/TaskList.java

```diff
@@ -0,0 +1,114 @@
+package muffin;
+
+import java.util.ArrayList;
+
+/**
+ * The TaskList class manages a list of tasks. It provides methods to add, delete, mark as done,
+ * unmark as done, and retrieve tasks, as well as to list all tasks or matching tasks in the list.
+ */
+public class TaskList {
+
+    /**
+     * The list of tasks.
+     */
+    ArrayList<Task> list;
```

Comment:
Maybe `tasks` would be a better name for this variable, to be more descriptive. Same for the `ArrayList<Task>` at line 95.